src/Microsoft.AspNetCore.WebUtilities/HttpRequestStreamReader.cs

```diff
@@ -132,9 +110,9 @@ protected override void Dispose(bool disposing)
 
         public override int Peek()
         {
-            if (_stream == null)
+            if (_disposed)
             {
-                throw new ObjectDisposedException("stream");
+                throw new ObjectDisposedException(nameof(HttpRequestStreamReader));
```

Comment:
ObjectDisposedException doesn't actually have a ctor with no arguments. We seem to do the nameof(classname) across our repos.